src/webots/nodes/WbLidar.hpp

```diff
@@ -137,7 +137,7 @@ class WbLidar : public WbAbstractCamera {
   bool isRotating() const { return mActualType.startsWith('r', Qt::CaseInsensitive); }
   double verticalFieldOfView() const { return actualFieldOfView() * ((double)height() / (double)width()); }
 
-  WbLidarPoint *pointArray() { return (WbLidarPoint *)(lidarImage() + actualHorizontalResolution() * actualNumberOfLayers()); }
+  WbLidarPoint *pointArray() { return (WbLidarPoint *)(lidarImage() + width() * height()); }
```

Comment:
That seems to fix #3220.

Reply:
It seems that the handling of the image is currently done like this:

- The `AbstractCamera` writes the data (`width` x `height`) in the shared memory (corresponding pointer is returned by`lidarImage()` in `WbLidar.cpp`)
- We copy it in a temporary image (`mTemporaryImage`, also `width` x `height`)
- We overwrite the shared memory by putting this time only the layers needed (`actualHorizontalResolution` x `actualNumberOfLayers`) for the point cloud.
- We compute the coordinates for the WbLidarPoint of the point cloud and save them in the shared memory just after the image --> so at `lidarImage() + width * height`

Reply:
The error when it was rotating is that the width of the image is not returned by `width() `but by `actualHorizontalResolution()`.